clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/DaVinciRecordTransformerUtility.java

```diff
@@ -146,25 +146,27 @@ public final void onRecovery(
     } else {
       // Bootstrap from local storage
       LOGGER.info("Bootstrapping from local storage for partition {}", partitionId);
-      AbstractStorageIterator iterator = storageEngine.getIterator(partitionId);
-      for (iterator.seekToFirst(); iterator.isValid(); iterator.next()) {
-        byte[] keyBytes = iterator.key();
-        byte[] valueBytes = iterator.value();
-        Lazy<K> lazyKey = Lazy.of(() -> keyDeserializer.deserialize(keyBytes));
-        Lazy<O> lazyValue = Lazy.of(() -> {
-          ByteBuffer valueByteBuffer = ByteBuffer.wrap(valueBytes);
-          // Skip schema id
-          valueByteBuffer.position(Integer.BYTES);
-          ByteBuffer decompressedValueBytes;
-          try {
-            decompressedValueBytes = compressor.get().decompress(valueByteBuffer);
-          } catch (IOException e) {
-            throw new RuntimeException(e);
-          }
-          return outputValueDeserializer.deserialize(decompressedValueBytes);
-        });
-
-        recordTransformer.processPut(lazyKey, lazyValue, partitionId);
+
+      try (AbstractStorageIterator iterator = storageEngine.getIterator(partitionId)) {
```

Comment:
When using iterator to scan the DB, it is better to put the DB as read-only mode to make sure there is no data missing as a safeguard.